vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -199,55 +206,172 @@ def __init__(
             params_dtype=params_dtype,
             weight_loader=self.weight_loader)
 
+    def _load_per_tensor_weight_scale(self, shard_id: str,
+                                      param: torch.nn.Parameter,
+                                      loaded_weight: torch.Tensor,
+                                      expert_id: int):
+        param_data = param.data
+        # for per tensor weight quantization
+        if shard_id in ("w1", "w3"):
+            # We have to keep the weight scales of w1 and w3 because
+            # we need to re-quantize w1/w3 weights after weight loading.
+            idx = 0 if shard_id == "w1" else 1
+            param_data[expert_id][idx] = loaded_weight
+        # If we are in the row parallel case (down_proj)
+        else:
```

Comment:
there's one exception at the very beginning of the weight_loader function. so it would be raised before we get to this point, if shard_id is not one of w1,w2,w3